src/Carousel.test.js

```diff
@@ -112,16 +112,16 @@ describe('Carousel', () => {
         await wait(() => expect(next.disabled).toBe(false));
 
         fireEvent.click(next);
-        await wait(() => expect(props.beforeChange).toHaveBeenNthCalledWith(1, { current: 0, next: 1 }));
-        await wait(() => expect(props.afterChange).toHaveBeenNthCalledWith(1, { previous: 0, current: 1 }));
+        await wait(() => expect(props.beforeChange).toHaveBeenNthCalledWith(1, { current: 0, next: 1, source: 'user' }));
+        await wait(() => expect(props.afterChange).toHaveBeenNthCalledWith(1, { previous: 0, current: 1, source: 'user' }));
 
         fireEvent.click(next);
-        await wait(() => expect(props.beforeChange).toHaveBeenNthCalledWith(2, { current: 1, next: 2 }));
-        await wait(() => expect(props.afterChange).toHaveBeenNthCalledWith(2, { previous: 1, current: 2 }));
+        await wait(() => expect(props.beforeChange).toHaveBeenNthCalledWith(2, { current: 1, next: 2, source: 'user' }));
+        await wait(() => expect(props.afterChange).toHaveBeenNthCalledWith(2, { previous: 1, current: 2, source: 'user' }));
 
         fireEvent.click(prev);
-        await wait(() => expect(props.beforeChange).toHaveBeenNthCalledWith(3, { current: 2, next: 1 }));
-        await wait(() => expect(props.afterChange).toHaveBeenNthCalledWith(3, { previous: 2, current: 1 }));
+        await wait(() => expect(props.beforeChange).toHaveBeenNthCalledWith(3, { current: 2, next: 1, source: 'user' }));
+        await wait(() => expect(props.afterChange).toHaveBeenNthCalledWith(3, { previous: 2, current: 1, source: 'user' }));
```

Comment:
There are a few tests for `autoplay`, can we add expectations for these callbacks to ensure the `source` is correctly set?

Reply:
Added a new test for that 👍